src/linalg/impl_linalg.rs

```diff
@@ -815,3 +815,132 @@ mod blas_tests {
         assert!(blas_column_major_2d::<f32, _>(&m));
     }
 }
+
+#[allow(dead_code)]
+fn general_outer_to_dyn<Sa, Sb, F, T>(
+    a: &ArrayBase<Sa, IxDyn>,
+    b: &ArrayBase<Sb, IxDyn>,
+    f: F,
+) -> ArrayD<T>
+where
+    T: Copy,
+    Sa: Data<Elem = T>,
+    Sb: Data<Elem = T>,
+    F: Fn(T, T) -> T,
+{
+    //Iterators on the shapes, compelted by 1s
+    let a_shape_iter = a.shape().iter().chain([1].iter().cycle());
+    let b_shape_iter = b.shape().iter().chain([1].iter().cycle());
+
+    let res_ndim = std::cmp::max(a.ndim(), b.ndim());
+    let res_dim: Vec<Ix> = a_shape_iter
+        .zip(b_shape_iter)
+        .take(res_ndim)
+        .map(|(x, y)| x * y)
+        .collect();
+
+    unsafe {
+        let mut res: ArrayD<T> = ArrayBase::uninitialized(res_dim);
```

Comment:
If you have time, this can be ported to `Array::maybe_uninit` now. Look into using `Zip::apply_assign_into` - see example in tests?  I'd still use `T: Copy` as the restriction, as long as we don't handle drop of partially filled arrays on panics.

Reply:
Nice!